game/game.py

```diff
@@ -437,16 +437,48 @@ def plan_procurement(self) -> None:
         # gets much more of the budget that turn. Otherwise budget (after
         # repairs) is split evenly between air and ground. For the default
         # starting budget of 2000 this gives 600 to ground forces and 1400 to
-        # aircraft.
-        ground_portion = 0.3 if self.turn == 0 else 0.5
+        # aircraft. After that the budget will be spend proportionally based on how much is already invested
+
+        player_armor_investment = 0
+        player_aircraft_investment = 0
+        player_ground_portion = 0.3
+
+        if self.turn is not 0:
+            for cp in self.theater.player_points():
+                player_armor_investment += cp.base.total_ground_units_value
+                player_aircraft_investment += cp.base.total_aircraft_value
```

Comment:
Both of these only track the currently available units, not those purchased or transferring. You need `ControlPoint.allocated_ground_units` and `ControlPoint.expected_aircraft_next_turn`.

I should probably see if I can rework `Base` at some point. It's mostly the container for raw data and calls into it should be fairly rare. A better name and more cohesion would probably help...

Reply:
expected_aircraft_next_turn only got int vals, not the dic's with the units in it, so I introduced a allocated_aircraft, and made the price calc there.
Not sure If this is the right place to do the price calc though, but the necessary  information is there.